controller/space_iterations_test.go

```diff
@@ -244,163 +245,141 @@ func (rest *TestSpaceIterationREST) TestFailListIterationsByMissingSpace() {
-	assert.NotEqual(rest.T(), uuid.UUID{}, iteration2.ID)
-
-	childOfIteration2 := iteration.Iteration{
-		Name:    "Sprint 2.1",
-		SpaceID: spaceInstance.ID,
-		Path:    append(iteration2.Path, iteration2.ID),
-	}
-	iterationRepo.Create(rest.Ctx, &childOfIteration2)
-	require.NotEqual(rest.T(), uuid.Nil, childOfIteration2.ID)
-
-	grandChildOfIteration2 := iteration.Iteration{
-		Name:    "Sprint 2.1.1",
-		SpaceID: spaceInstance.ID,
-		Path:    append(childOfIteration2.Path, childOfIteration2.ID),
-	}
-	iterationRepo.Create(rest.Ctx, &grandChildOfIteration2)
-	require.NotEqual(rest.T(), uuid.UUID{}, grandChildOfIteration2.ID)
-
-	wirepo := workitem.NewWorkItemRepository(rest.DB)
-
-	for i := 0; i < 3; i++ {
-		wirepo.Create(
-			rest.Ctx, iteration1.SpaceID, workitem.SystemBug,
-			map[string]interface{}{
-				workitem.SystemTitle:     fmt.Sprintf("New issue #%d", i),
-				workitem.SystemState:     workitem.SystemStateNew,
-				workitem.SystemIteration: iteration1.ID.String(),
-			}, rest.testIdentity.ID)
-	}
-	for i := 0; i < 2; i++ {
-		_, err := wirepo.Create(
-			rest.Ctx, iteration1.SpaceID, workitem.SystemBug,
-			map[string]interface{}{
-				workitem.SystemTitle:     fmt.Sprintf("Closed issue #%d", i),
-				workitem.SystemState:     workitem.SystemStateClosed,
-				workitem.SystemIteration: iteration1.ID.String(),
-			}, rest.testIdentity.ID)
-		require.Nil(rest.T(), err)
-	}
-	// add items to nested iteration level 1
-	for i := 0; i < 4; i++ {
-		_, err := wirepo.Create(
-			rest.Ctx, iteration1.SpaceID, workitem.SystemBug,
-			map[string]interface{}{
-				workitem.SystemTitle:     fmt.Sprintf("New issue #%d", i),
-				workitem.SystemState:     workitem.SystemStateNew,
-				workitem.SystemIteration: childOfIteration2.ID.String(),
-			}, rest.testIdentity.ID)
-		require.Nil(rest.T(), err)
-	}
-	// add items to nested iteration level 2
-	for i := 0; i < 5; i++ {
-		_, err := wirepo.Create(
-			rest.Ctx, iteration1.SpaceID, workitem.SystemBug,
-			map[string]interface{}{
-				workitem.SystemTitle:     fmt.Sprintf("Closed issue #%d", i),
-				workitem.SystemState:     workitem.SystemStateClosed,
-				workitem.SystemIteration: grandChildOfIteration2.ID.String(),
-			}, rest.testIdentity.ID)
-		require.Nil(rest.T(), err)
-	}
+	fxt := tf.NewTestFixture(rest.T(), rest.DB,
+		tf.CreateWorkItemEnvironment(),
+		tf.Iterations(5,
+			tf.SetIterationNames("root", "Sprint 1", "Sprint 2", "Sprint 2.1", "Sprint 2.1.1"),
+			func(fxt *tf.TestFixture, idx int) error {
+				i := fxt.Iterations[idx]
+				switch idx {
+				case 1:
+					i.MakeChildOf(*fxt.Iterations[0])
+				case 2:
+					i.MakeChildOf(*fxt.Iterations[0])
+				case 3:
+					i.MakeChildOf(*fxt.Iterations[2])
+				case 4:
+					i.MakeChildOf(*fxt.Iterations[3])
+				}
+				return nil
+			}),
+
+		tf.WorkItems(28, func(fxt *tf.TestFixture, idx int) error {
+			wi := fxt.WorkItems[idx]
+			switch idx {
+			case 0, 1, 2:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateNew
```

Comment:
@baijum  FYI, did you know that there's also a function to customize a work item by field?
```go
tf.WorkItems(2, tf.SetWorkItemField(workitem.SystemTitle, "wi", "wi2"))
```
Not sure if it is particularly helpful here though except for state.